### ISSUE TITLE
Issue when installing pachi-py

### ISSUE THREAD

**Issue by RajeshMaapu** [openai/pachi-py#6]
Hi, This is the error i get when installing pachi-py separately. Can you please give any insight ?

[pachiError.pdf](https://github.com/openai/pachi-py/files/522535/pachiError.pdf)

**Comment by tlbtlbtlb** [comment]
One error is here:

```
CMake Error at /usr/local/Cellar/cmake/3.5.2/share/cmake/Modules/CMakeDetermineCompilerId.cmake:132 (file):
  file failed to open for writing (Permission denied):
    /Users/rajesh/pachi-py/pachi_py/build/CMakeFiles/3.5.2/CompilerIdC/CMakeCCompilerId.c
```

Perhaps these files are owned by root or another user? In your pachi-py directory, `sudo chown -R rajesh .` should fix it.

Another error is here:

```
CMAKE_OSX_DEPLOYMENT_TARGET is '10.12' but CMAKE_OSX_SYSROOT:
   ""
  is not set to a MacOSX SDK with a recognized version.  Either set
  CMAKE_OSX_SYSROOT to a valid SDK or set CMAKE_OSX_DEPLOYMENT_TARGET
to empty
```

Also:

```
➜ pachi-py git:(master) ✗ DCMAKE_OSX_SYSROOT=/path/to/Mac10.12SDK
```

Are you literally typing in `/path/to/`? You have to replace that with the actual path on your system. And I'm not sure what the `D` is for. The command you probably want is:

```
export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/ Developer/SDKs/MacOSX10.12.sdk
```

**Comment by RajeshMaapu** [comment]
Hi, Thank you, The first issue is fixed with the above command. I was earlier trying with dsenableroot command to get rid of it. Thank you very much.

And the 2nd error log, is still coming while installation.

CMAKE_OSX_DEPLOYMENT_TARGET is '10.12' but CMAKE_OSX_SYSROOT:
   ""
  is not set to a MacOSX SDK with a recognized version.  Either set
  CMAKE_OSX_SYSROOT to a valid SDK or set CMAKE_OSX_DEPLOYMENT_TARGET
to empty
[uploadPachi.pdf](https://github.com/openai/pachi-py/files/523170/uploadPachi.pdf)

I tried the above suggestion to use the actual path in CMAKE_OSX_SYSROOT command and also used export SDKROOT command as given above. The complete log file is attached for your guidance. Thank you.

**Comment by tlbtlbtlb** [comment]
```
bash-3.2$ CMAKE_OSX_SYSROOT=/Applications/Xcode.app/Contents/
Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk
```

You left out the `export` at the beginning -- without that it has no effect.

**Comment by RajeshMaapu** [comment]
@tlbtlbtlb Thanks for responding, I rectified that. Though I got same error. I'm not sure what to make of it now. 
[pachi_2_pdf.pdf](https://github.com/openai/pachi-py/files/523214/pachi_2_pdf.pdf)

**Comment by RajeshMaapu** [comment]
I think I moved closer with this below error log, I got rid of cmake error (somehow) Used unset CMAKE_OSX_DEPLOYMENT_TARGET and xcrun make command. Please let me know if I'm in the right direction and if this new error can be cleared 'clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command 'g++' failed with exit status 1'
[closerError.pdf](https://github.com/openai/pachi-py/files/523529/closerError.pdf)

bash-3.2$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.38)
Target: x86_64-apple-darwin16.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
bash-3.2$

**Comment by tlbtlbtlb** [comment]
It may take some experimentation to get Pachi working on OSX 10.12. Are you actually planning to use Pachi to develop Go-playing algorithms in the next few weeks? If not, I'd suggest installing gym without it and and waiting until 10.12 is better supported.

If you do want to try to make Pachi work on 10.12, you might look at how other packages made it work. For example, http://forum.openscenegraph.org/viewtopic.php?t=16028 or https://github.com/search?q=CMAKE_OSX_SYSROOT+to+a+valid+SDK&type=Issues&utf8=%E2%9C%93

**Comment by RajeshMaapu** [comment]
No, I'm just looking to work on text environment. May I know how to install gym without pachi ? And also my recent error log is : 'clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: command 'g++' failed with exit status 1'
Do we also think it's related to cmake incompatibility with OSX 10.12 or this is a different issue?

**Comment by tlbtlbtlb** [comment]
https://github.com/openai/gym#installation

Yes, the linker is failing because it's trying to link to support the ancient OSX 10.6, which doesn't support some features Pachi requires. On 10.11, we can just set MACOS_DEPLOYMENT_TARGET=10.11 and it works. I don't know what changed on 10.12.

**Comment by RajeshMaapu** [comment]
Thank you. I'll close this case. I did the following..
added unset(CMAKE_OSX_DEPLOYMENT_TARGET) in the cmake file. 
didn't use "export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk" (or 10.11.sdk) when installation in terminal as suggested in other solutions.
Used MACOS_DEPLOYMENT_TARGET=10.11 alone when installation on terminal. 
Gym installed successfully in OS 10.12 after that. (No Pachi errors)

Added point, once installed new version CLT, run Xcode. 

A rookie on shell commands and understanding of the errors dragged me this issue.

Thank you @tlbtlbtlb , Interacting with you helped me gain some knowledge and insight helped solving this issue. Owe you a ton.
